cmd/backup-manager/app/util/util.go

```diff
@@ -431,3 +432,22 @@ func GetContextForTerminationSignals(op string) (context.Context, context.Cancel
 	}()
 	return ctx, cancel
 }
+
+// RetryOnError allows the caller to retry fn in case the error returned by fn.
+// sleep define the interval between two retries.
+func RetryOnError(attempts int, sleep time.Duration, fn func() error) error {
+	var lastErr error
+	for i := 0; i < attempts; i++ {
+		err := fn()
+		if err == nil {
+			return nil
+		}
+
+		lastErr = err
+		if sleep != 0 {
+			time.Sleep(sleep)
```

Comment:
do we need to support cancel with a context?

Reply:
add a callback `retriable func(error) bool` to support cancel, so we can cancel with context in `retriable`.

Reply:
but we `RetriableOnAnyError` now?

Reply:
if the context is done, the delete function will fail fastly, so I think it is ok.

Reply:
Updated